service/fullnode/fullnode/fullnode.go

```diff
@@ -0,0 +1,221 @@
+package fullnode
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/zeromicro/go-zero/core/logx"
+	"gorm.io/gorm"
+
+	"github.com/bnb-chain/zkbnb-go-sdk/client"
+	"github.com/bnb-chain/zkbnb/core"
+	"github.com/bnb-chain/zkbnb/dao/block"
+	tx "github.com/bnb-chain/zkbnb/dao/tx"
+	"github.com/bnb-chain/zkbnb/types"
+)
+
+const (
+	MaxFullnodeInterval = 60 * 1
+	DefaultL2EndPoint   = "http://localhost:8888"
+	SyncInterval        = 100 * time.Millisecond
+)
+
+type Config struct {
+	core.ChainConfig
+
+	ApiServer struct {
+		L2EndPoint string
+	}
+	FullNode struct {
+		SyncBlockStatus int64
+	}
+
+	LogConf logx.LogConf
+}
+
+type Fullnode struct {
+	config *Config
+	client client.ZkBNBClient
+	bc     *core.BlockChain
+
+	quitCh chan struct{}
+}
+
+func NewFullnode(config *Config) (*Fullnode, error) {
+	bc, err := core.NewBlockChain(&config.ChainConfig, "fullnode")
+	if err != nil {
+		return nil, fmt.Errorf("new blockchain error: %v", err)
+	}
+
+	l2EndPoint := config.ApiServer.L2EndPoint
+	if len(l2EndPoint) == 0 {
+		l2EndPoint = DefaultL2EndPoint
+	}
+
+	if config.FullNode.SyncBlockStatus <= block.StatusProposing ||
+		config.FullNode.SyncBlockStatus > block.StatusVerifiedAndExecuted {
+
+		config.FullNode.SyncBlockStatus = block.StatusVerifiedAndExecuted
+	}
+
+	fullnode := &Fullnode{
+		config: config,
+		client: client.NewZkBNBClient(l2EndPoint),
+		bc:     bc,
+
+		quitCh: make(chan struct{}),
+	}
+	return fullnode, nil
+}
+
+func (c *Fullnode) Run() {
+	// TODO: add BlockSize in zkbnb-go-sdk
+
+	curHeight, err := c.bc.BlockModel.GetCurrentBlockHeight()
+	if err != nil {
+		panic(fmt.Sprintf("get current block height failed, error: %v", err.Error()))
+	}
+
+	curBlock, err := c.bc.BlockModel.GetBlockByHeight(curHeight)
+	if err != nil {
+		panic(fmt.Sprintf("get current block failed, height: %d, error: %v", curHeight, err.Error()))
+	}
+
+	ticker := time.NewTicker(SyncInterval)
+	defer ticker.Stop()
+
+	syncBlockStatus := c.config.FullNode.SyncBlockStatus
+	for {
+		select {
+		case <-ticker.C:
+
+			// if the latest block have been created
+			if curBlock.BlockStatus > block.StatusProposing {
+				// init new block, set curBlock.status to block.StatusProposing
+				curBlock, err = c.bc.InitNewBlock()
+				if err != nil {
+					panic(fmt.Sprintf("init new block failed, block height: %d, error: %v", curHeight, err.Error()))
+				}
+
+				curHeight++
+			}
+
+			l2Block, err := c.client.GetBlockByHeight(curHeight)
+			if err != nil && err != types.DbErrNotFound {
+				logx.Errorf("get block failed, height: %d, err %v ", curHeight, err)
```

Comment:
I think all err should be handled here, it should like this:

if err != nil {
    if err != types.DbErrNotFound {
         logx.Errorf("***")
    }
    continue
}